# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.cpp
# Commit: b94def6b6a8b
# Full Hash: b94def6b6a8b846b166726c84f39161e656d3b41
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-03-02 21:22:31
# Description:
#   Bug 1819742 - Directly set up stream according to the IPC stream type in InternalRequest. r=dom-worker-reviewers,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171450
# ==============================================================================

diff -r 3b2720a71b77 -r b94def6b6a8b dom/fetch/InternalRequest.cpp
--- a/dom/fetch/InternalRequest.cpp	Thu Mar 02 19:26:01 2023 +0200
+++ b/dom/fetch/InternalRequest.cpp	Thu Mar 02 17:44:45 2023 +0000
@@ -190,14 +190,11 @@
 
   const Maybe<BodyStreamVariant>& body = aIPCRequest.body();
 
-  if (!XRE_IsParentProcess()) {
-    if (body) {
-      MOZ_ASSERT(body->type() == BodyStreamVariant::TParentToChildStream);
+  if (body) {
+    if (body->type() == BodyStreamVariant::TParentToChildStream) {
       mBodyStream = body->get_ParentToChildStream().get_RemoteLazyInputStream();
     }
-  } else {
-    if (body) {
-      MOZ_ASSERT(body->type() == BodyStreamVariant::TChildToParentStream);
+    if (body->type() == BodyStreamVariant::TChildToParentStream) {
       mBodyStream =
           DeserializeIPCStream(body->get_ChildToParentStream().stream());
     }
